livespark-form-modeler/livespark-form-modeler-codegen/livespark-form-modeler-codegen-impl/src/test/java/org/livespark/formmodeler/codegen/flow/impl/RoasterFlowProducerSourceGeneratorTest.java

```diff
@@ -0,0 +1,188 @@
+    private org.guvnor.common.services.project.model.Package root;
+    private org.guvnor.common.services.project.model.Package local;
+    private org.guvnor.common.services.project.model.Package shared;
+    private List<FormDefinition> projectForms;
+    private SourceGenerationContext context;
+
+    @Before
+    public void setUp() {
+        sourceGenerator = new RoasterFlowProducerSourceGenerator();
+
+        formDefinition = new FormDefinition();
+        formDefinition.setId(FORM_ID);
+        formDefinition.setName(FORM_NAME);
+
+        formDefinition.setModel(new DataObjectFormModel("employee", "org.juraj.mypackage.Employee"));
+
+        path = PathFactory.newPath(FORM_NAME, FORM_URI);
+        shared = new org.guvnor.common.services.project.model.Package(path, null, null, null, null, PACKAGE, "caption", "relativeCaption");
+        local = new org.guvnor.common.services.project.model.Package(path, null, null, null, null, PACKAGE, "caption", "relativeCaption");
+
+        context = new SourceGenerationContext(formDefinition, path, root, local, shared, null, projectForms);
+        generatedSource = sourceGenerator.generateJavaSource(context);
+        gslen = generatedSource.length();
+    }
+
+    @Test
+    public void generateJavaSource() {
+        assertThat(getImports()).containsAll(IMPORTS);
+
+        assertThat(getPackage()).isEqualTo(PACKAGE);
+        assertThat(getClassAnnotation()).isEqualTo(CLASS_ANNOTATION);
+        assertThat(getFullClassName()).isEqualTo(FULL_CLASS_NAME);
+
+        assertThat(getAbstractMethodsImplementations()).isEqualTo(ABSTRACT_METHODS_IMPLEMENTATIONS);
+        assertThat(getProducerMethodsImplementations()).isEqualTo(PRODUCER_METHODS_IMPLEMENTATIONS);
+
+        Map<String, List<String>> methodsAnnotations = getAllMethodsAnnotations();
+        for (String abstractMethod : ABSTRACT_METHODS) {
+            assertThat(methodsAnnotations.get(abstractMethod)).isEqualTo(Arrays.asList("Override"));
+        }
+        for (String producerMethod : PRODUCER_METHODS.subList(1, 5)) {
+            assertThat(methodsAnnotations.get(producerMethod)).containsExactlyElementsOf(Arrays.asList("Override", "Produces", "Singleton", "ForEntity(\"org.juraj.mypackage.myFormName\")", "Named(\"" + producerMethod + "\")"));
+        }
+    }
+
+    private Set<String> getImports() {
+        Set<String> imports = new HashSet<>();
+        int start = generatedSource.indexOf("import") + 6;
+        if (start != 5) {
+            int end;
+            char nextChar;
+            StringBuilder nextImport;
+            do {
+                nextImport = new StringBuilder();
+                end = start + 1;
+                do {
+                    nextChar = generatedSource.charAt(end++);
+                    if (nextChar != ';') {
+                        nextImport.append(nextChar);
+                    } else {
+                        break;
+                    }
+                } while (true);
```

Comment:
This loop should be separated out into a new function, something like `findNextChar(where, what, startAt)`. Or better yet, `String.indexOf()`.